server/service/src/static_files.rs

```diff
@@ -135,6 +149,130 @@ impl StaticFileService {
             path: file_path.to_string_lossy().to_string(),
         }))
     }
+
+    pub async fn download_file_from_central(
+        &self,
+        id: &str,
+        category: StaticFileCategory,
+        service_provider: &ServiceProvider,
+    ) -> anyhow::Result<Option<StaticFile>> {
+        log::info!("Downloading file from central server");
+        if is_central_server() {
+            return Err(anyhow::Error::msg(
+                "Can't download file from central server, as I am the central server!",
+            ));
+        }
+
+        let (table_name, record_id) = match category.clone() {
+            StaticFileCategory::SyncFile(table_name, record_id) => (table_name, record_id),
```

Comment:
I feel like the verboseness of rust code can really get in a way of understanding logic, i like using Let/else and ok_or, etc to try and reduce it a little bit, for example: https://github.com/msupply-foundation/open-msupply/compare/3398-file-sync-upload-to-central-server...3398-file-sync-upload-to-central-server(Less-verbose-mapping)

Reply:
Also would this method ever be called with StatiFileCateogry that is not sync_file ? Maybe sync file needs to a be a type that is stored in StaticFileCategory::SyncFile ? And we can pass that type here explicitly then construt StatiFileCategory if we need to reserver file etc..